htcanalyze/main.py

```diff
@@ -562,20 +568,25 @@ def print_results(htcanalyze: HTCAnalyze,
         else:
             results = htcanalyze.analyzed_summary(log_files)  # else
 
-    # Allow this to happen
+    # Allow to end program without output, if no scripts were analyzed
     if results is None:
         sys.exit(0)
 
-    work_with = results
     # convert result to list, if given as dict
     if isinstance(results, dict):
         work_with = [results]
+    else:
+        work_with = results
 
     # check for ignore values
+    # TODO: I think, "mystery" should be named "result"?
+    # In any case, this name should describe at least something!
     for mystery in work_with:
 
         for i in mystery:
```

Comment:
@mathisloevenich If mystery is an iterable, call it `mysteries`. Well, I'm not telling you to use `mystery` or `mysteries` as a name, I'm telling you to use plurals. Do not use `i`, because it's not a counter.

The way you seem to use `i` is as an element-id inside mystery. I don't quite understand what you're trying to do here. There's obviously some problem you're trying to tackle, but there are no comments nor are the variables descriptive in any way. Does `mystery` contain element-ids that reference itself?

Can you please explain this?

Reply:
Well you caught me in my laziness. Of cause this all needs better naming. When I created the code, I did not know what to name it:
For explanation: 
- work_with is a list of dicts containing the results of the *processed data* by any mode, so for ex. `processed_data_list` would match it
- mystery are the different dictionaries, containing data like: times, resources, execution-details, ...
- the iterating `for i in mystery` is to check if one of the key: value pairs contains a *None* element, which is not supposed to happen. So this section is more or less for the developer and I did forget to remove/improve it.

Reply:
My suggestions:
- keep `results` in lieu of `work_with`
- loop over `results` as `for result in results`
- rename `mystery` to `entries` or some other descriptive plural name for those dictionaries

Since a key cannot be `None`, the best way of checking would be:

```
for entry in entries:
    if not all(entry.values()):
        logging.debug('There are None/False/0 values in the dictionary')
```

However, it seems like this piece of code is dedicated to debugging / testing. Why not turn this into a test function for your unit testing? You can then assert whatever you expect _should_ happen, and if it doesn't the testing suite will tell you where it went wrong. You can be as specific with your asserts as you like. You can make one to ensure there are no NoneType values in there, and you can also make an assert to check whether a value is like you expect it to be (i.e., `isinstance(value, int)` for integers, or even specific values when you use dummy input).